packages/domain/source/server/appeals/appeal.js

```diff
@@ -42,31 +44,47 @@ Space.eventSourcing.Aggregate.extend(Donations, `Appeal`, {
   },
 
   _makePledge(command) {
-    // Pledges can only be made for open appeals.
-    if (this.hasState(this.STATES.fulfilled)) {
-      throw new Donations.PledgeCannotBeMadeToFulfilledAppeal();
-    }
-    // Pledges are capped at the appeal’s required quantity
-    quantity = command.quantity;
-    newPledgedQuantity = this.pledgedQuantity.add(quantity);
-    if (newPledgedQuantity.isMore(this.requiredQuantity)) {
-      quantity = quantity.substract(newPledgedQuantity.delta(this.requiredQuantity));
-      command.quantity = quantity; // Assign capped quantity
-    }
-    pledgedQuantity = this.pledgedQuantity.add(quantity);
-    this.record(new Donations.PledgeMade(this._eventPropsFromCommand(command)));
-    // An appeal is fulfilled when the sum of pledged items equals the required quantity.
-    if (pledgedQuantity.equals(this.requiredQuantity)) {
-      this.record(new Donations.AppealFulfilled({ sourceId: this.getId() }));
+    if (this.hasState(this.STATES.open)) {
+      // Pledged quantity is capped at the appeal’s required quantity
+      let quantity = command.quantity;
+      let newPledgedQuantity = this.pledgedQuantity.add(quantity);
+      if (newPledgedQuantity.isMore(this.requiredQuantity)) {
+        quantity = quantity.substract(newPledgedQuantity.delta(this.requiredQuantity));
+        command.quantity = quantity; // Assign capped quantity
+      }
+      let pledgedQuantity = this.pledgedQuantity.add(quantity);
+      this.record(new Donations.PledgeMade(this._eventPropsFromCommand(command)));
+      // Fulfilled when the sum of pledged items equals the required quantity.
+      if (pledgedQuantity.equals(this.requiredQuantity)) {
+        this.record(new Donations.AppealFulfilled({
+          sourceId: this.getId(),
+          title: this.title,
+          requiredQuantity: this.requiredQuantity,
+          organizationId: this.organizationId,
+          locationId: this.locationId,
+          description: this.description
+        }));
+      }
+    } else {
+      throw new Donations.AppealNotOpenForNewPledges();
```

Comment:
Please throw early instead of wrapping everything into an `if` clause. So i would just turn this around -> if not has state open then throw immediately at the beginning.